src/main/scala/uk/gov/hmrc/apiplatform/modules/apis/domain/models/ExtendedApiDefinition.scala

```diff
@@ -23,17 +23,40 @@ import uk.gov.hmrc.apiplatform.modules.common.domain.models._
 import uk.gov.hmrc.apiplatform.modules.common.domain.services.InstantJsonFormatter
 
 case class ExtendedApiDefinition(
-    serviceName: String, // TODO : Should be ServiceName but breaking change to APM
+    serviceName: ServiceName,
     serviceBaseUrl: String,
     name: String,
     description: String,
     context: ApiContext,
+    versions: List[ExtendedApiVersion],
     requiresTrust: Boolean,
     isTestSupport: Boolean,
-    versions: List[ExtendedApiVersion],
-    categories: List[ApiCategory] = List.empty,
-    lastPublishedAt: Option[Instant]
-  )
+    lastPublishedAt: Option[Instant],
+    categories: List[ApiCategory] = List.empty
+  ) {
+
+  def userAccessibleApiDefinition = {
+    copy(versions =
+      versions.filter(v =>
+        v.productionAvailability.exists(_.isAccessible) || v.sandboxAvailability.exists(_.isAccessible)
+      )
+    )
+  }
+
+  val statusOrdering: Ordering[ApiStatus] = Ordering.by[ApiStatus, Int](ApiStatus.priorityOf)
+
+  val statusExtendedVersionOrdering: Ordering[ExtendedApiVersion] = Ordering.by[ExtendedApiVersion, ApiStatus](_.status)(statusOrdering).reverse
```

Comment:
Should this be reversed in api-documentation-frontend rather than here?

Reply:
Don't think so as this is the order defined by the library. It using a reverse is opaque to the outside.